python_modules/dagster-graphql/dagster_graphql/schema/evaluate_sensor.py

```diff
@@ -0,0 +1,82 @@
+from typing import TYPE_CHECKING, Any, Mapping
+
+import dagster._check as check
+import graphene
+from dagster._core.definitions.schedule_definition import ScheduleExecutionData
+from dagster._core.definitions.sensor_definition import SensorExecutionData
+from dagster._core.host_representation.selector import SensorSelector
+
+from dagster_graphql.implementation.utils import capture_error
+
+from ..implementation.evaluate_sensor_utils import evaluate_sensor
+from ..schema.errors import GrapheneSensorNotFoundError
+from .errors import GraphenePythonError
+from .inputs import GrapheneSensorSelector
+from .instigation import GrapheneRunRequest
+
+if TYPE_CHECKING:
+    from dagster_graphql.schema.util import HasContext
+
+
+class GrapheneSensorExecutionData(graphene.ObjectType):
+    cursor = graphene.String()
+    run_requests = graphene.List(GrapheneRunRequest)
+    skip_message = graphene.String()
+
+    class Meta:
+        name = "SensorExecutionData"
+
+    def __init__(self, execution_data: SensorExecutionData):
+        super().__init__()
+        self._execution_data = check.inst_param(
+            execution_data, "execution_data", SensorExecutionData
+        )
+
+    def resolve_cursor(self, _graphene_info: "HasContext"):
+        # Not present for schedules
+        if isinstance(self._execution_data, ScheduleExecutionData):
+            raise Exception("Attempted to retrieve cursor from schedule execution.")
+        return self._execution_data.cursor
+
+    def resolve_runRequests(self, _graphene_info: "HasContext"):
+        run_requests = self._execution_data.run_requests or []
+        return [GrapheneRunRequest(run_request) for run_request in run_requests]
+
+    def resolve_skipMessage(self, _graphene_info: "HasContext"):
+        return self._execution_data.skip_message
+
+
+class GrapheneEvaluateSensorResult(graphene.Union):
+    class Meta:
+        types = (
+            GrapheneSensorExecutionData,
+            GraphenePythonError,
+            GrapheneSensorNotFoundError,
+        )
+        name = "EvaluateSensorMutation"
```

Comment:
`EvaluateSensorResult` **

This caused me quite a bit of pain... the type generation quietly produces unusable output because of the collision of `EvaluateSensorMutation` being used two name this result but also the mutation.

Reply:
agh thanks for the catch - will update